### PR TITLE
SLING-8182 clarify servlet resolution order

### PR DESCRIPTION
**Pull request by kwin**
also explain servlet name property
fix some typos

### CODE REVIEW

**Comment by raducotescu** [comment]
@kwin, this looks ok to me. I'd rephrase a bit some sentences, but all the key information is correct. Would you mind if I merge this and #31? They're related IMO.

**Comment by kwin** [comment]
@raducotescu Thanks for the review. I merged already. Feel free to rephrase a bit with follow-up commits.
